File: Project 7/Java Version/TreeList.java
import java.io.FileWriter;

public class TreeList {

	private TreeNode listHead;

	TreeList() {
		this.listHead = new TreeNode();
	}
	
	public TreeNode getListHead() {
		return listHead;
	}
	
	public void setListHead(TreeNode listHead) {
		this.listHead = listHead;
	}

	public void insertNewNode(TreeNode head, TreeNode newNode) {
		TreeNode spot = findSpot(head, newNode);
		insert(spot, newNode);
	}

	public TreeNode findSpot(TreeNode head, TreeNode newNode) {
		TreeNode spot = head;
		while (spot.getNext() != null && spot.getNext().getProbability() < newNode.getProbability()) {
			spot = spot.getNext();
		}
		return spot;
	}

	public void insert(TreeNode spot, TreeNode newNode) {
		newNode.setNext(spot.getNext());
		spot.setNext(newNode);
	}

	public void printList(TreeNode head, FileWriter output) {
		
		String str = "listHead --> ";
		
		try {
			
			for(TreeNode node = head; node.getNext() != null; node = node.getNext()) {
				str += node.printNode(node);
			}
			str += "NULL\n";
			output.write(str);

		}
		catch (Exception e) {
			e.printStackTrace();
		}
	}
}

File: Project 7/C++ Version/TreeList.cpp
#include "TreeList.h"
#include <iostream>


TreeList::TreeList() {
	this->listHead = new TreeNode();
}

TreeList::~TreeList() {

}

TreeNode* TreeList::getListHead() {
	return this->listHead;
}

void TreeList::setListHead(TreeNode* listHead) {
	this->listHead = listHead;
}

void TreeList::insertNewNode(TreeNode* head, TreeNode* newNode) {
	TreeNode* spot = findSpot(head, newNode);
	insert(spot, newNode);
}

TreeNode* TreeList::findSpot(TreeNode* head, TreeNode* newNode) {
	TreeNode* spot = head;
	while (spot->getNext() != NULL && spot->getNext()->getProbability() < newNode->getProbability()) {
		spot = spot->getNext();
	}
	return spot;
}

void TreeList::insert(TreeNode* spot, TreeNode* newNode) {
	newNode->setNext(spot->getNext());
	spot->setNext(newNode);
}

void TreeList::printList(TreeNode* head, ofstream& output) {
	string str = "listHead --> ";
	try {
		for (TreeNode* node = head; node->getNext() != NULL; node = node->getNext()) {
			str += node->printNode();
		}
		str += "NULL\n";
		output << str;
	}
	catch (exception e) {
		cout << e.what();
	}
}